api/oci/extensions/repositories/ocireg/utils.go

```diff
@@ -38,7 +39,8 @@ func NewDataAccess(repo registry.Repository, digest digest.Digest, delayed bool)
 	var reader io.ReadCloser
 	desc, err := repo.Resolve(dummyContext, digest.String())
 	if err != nil {
-		if errors.Is(err, errdef.ErrNotFound) {
+		// docker registry sometimes returns an Internal Server Error instead of NOT FOUND!
+		if errors.Is(err, errdef.ErrNotFound) || strings.Contains(err.Error(), "Internal Server Error") {
```

Comment:
This feels like a hack for a proper retry, wdyt?

Reply:
There is no retry. It's not an error that can go away. Docker registry replies with an internal error when you are trying to find a blob that doesn't exist.

Reply:
Is this just a docker.io thing? If so should we maybe set this up explicitly for that registry? because all 500 feel a little strange.